airflow-core/src/airflow/ui/tests/e2e/specs/home-dashboard.spec.ts

```diff
@@ -0,0 +1,137 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { expect, test } from "@playwright/test";
+import { HomePage } from "tests/e2e/pages/HomePage";
+
+test.describe("Dashboard Metrics Display", () => {
+  let homePage: HomePage;
+
+  test.beforeEach(({ page }) => {
+    homePage = new HomePage(page);
+  });
+
+  test("should display dashboard stats section with DAG metrics", async () => {
+    await homePage.navigate();
+    await homePage.waitForDashboardLoad();
+
+    // Use Playwright assertions directly for clearer error messages
+    await expect(homePage.statsSection).toBeVisible();
+
+    await expect(homePage.activeDagsCard).toBeVisible();
+    const activeDagsCount = await homePage.getActiveDagsCount();
+
+    expect(activeDagsCount).toBeGreaterThanOrEqual(0);
+
+    await expect(homePage.runningDagsCard).toBeVisible();
+    const runningDagsCount = await homePage.getRunningDagsCount();
+
+    expect(runningDagsCount).toBeGreaterThanOrEqual(0);
+
+    await expect(homePage.failedDagsCard).toBeVisible();
+    const failedDagsCount = await homePage.getFailedDagsCount();
+
+    expect(failedDagsCount).toBeGreaterThanOrEqual(0);
+  });
+
+  test("should display health status badges", async () => {
+    await homePage.navigate();
+    await homePage.waitForDashboardLoad();
+
+    // Use Playwright assertions directly for clearer error messages
+    await expect(homePage.healthSection).toBeVisible();
+    await expect(homePage.metaDatabaseHealth).toBeVisible();
+    await expect(homePage.schedulerHealth).toBeVisible();
+    await expect(homePage.triggererHealth).toBeVisible();
+  });
+
+  test("should navigate to filtered DAGs list when clicking stats cards", async () => {
+    await homePage.navigate();
+    await homePage.waitForDashboardLoad();
+
+    await homePage.activeDagsCard.click();
+    await homePage.page.waitForURL(/paused=false/);
+
+    expect(homePage.page.url()).toContain("paused=false");
+
+    await homePage.navigate();
+    await homePage.waitForDashboardLoad();
+
+    await homePage.runningDagsCard.click();
+    await homePage.page.waitForURL(/last_dag_run_state=running/);
+
+    expect(homePage.page.url()).toContain("last_dag_run_state=running");
+  });
+
+  test("should display welcome heading on dashboard", async () => {
+    await homePage.navigate();
+    await homePage.waitForDashboardLoad();
+
+    await expect(homePage.welcomeHeading).toBeVisible();
+  });
+
+  test("should refresh and display updated metrics on page reload", async () => {
```

Comment:
We have to check if metrics update when DAGs are triggered/completed. Current test just reload page and check if metric is still available

```suggestion
  test("should update metrics when DAG is triggered", async () => {
  await homePage.navigate();
  await homePage.waitForDashboardLoad();

  const initialRunningCount = await homePage.getRunningDagsCount();

  const dagsPage = new DagsPage(homePage.page);
  await dagsPage.triggerDag(testConfig.testDag.id);

  await homePage.navigate();
  await homePage.waitForDashboardLoad();

});
```

Reply:
Thank you for pointing that ,addressed it